### ISSUE TITLE
Is there a built-in way to get a ticking sounds timer during work interval

### ISSUE THREAD

**Issue by meliache** [alphapapa/hammy.el#4]
This is something I got used to from `org-pomodoro`. My question is whether I can implement a ticking-sound timer that plays during the work interva in an elegant wayl using special hammy utilities like the `remind` function. I manager to add such a functionality to a hammy myself, but my solution looks very crude to me and I think there must be a better way. Also, if there is an easy way to do this, it might be worth it adding it to some documentation, either to a readme example or to some github wiki or something like that.

My solution was using a custom timer and functions for starting and cancelling it,
``` emacs-lisp
  (require 'org-pomodoro)
  (defun my-play-ticking-sound ()
    (org-pomodoro-play-sound ' :tick))

  (defvar my-ticking-sound-timer nil)
  (defun my-ticking-sound-start-timer ()
    (interactive)
    (if (not my-ticking-sound-timer)
        (setf my-ticking-sound-timer
              (run-at-time nil 2 #'my-play-ticking-sound))))

  (defun my-ticking-sound-cancel-timer ()
    (interactive)
    (when (timerp my-ticking-sound-timer)
      (cancel-timer my-ticking-sound-timer))
    (setf my-ticking-sound-timer nil))
```
 and then I just calling the start- and stopping-functions in the `:before` and `:advance` hammy sections, respectively:
 
 ``` emacs-lisp
   (hammy-define (propertize "🍅" 'face '(:foreground "tomato"))
    :documentation "The classic pomodoro timer."
    :intervals
    (list
     (interval :name "Work"
               :duration "25 minutes"
               :before (do (announce "Starting work time.")
                           (notify "Starting work time.")
                           (org-pomodoro-play-sound ' :start)
                           (my-ticking-sound-start-timer))
               :advance (do (announce "Break time!")
                            (notify "Break time!")
                            (org-pomodoro-play-sound ' :short-break)
                            (my-ticking-sound-cancel-timer)))
     (interval :name "Break"
               :duration (do (if (and (not (zerop cycles))
                                      (zerop (mod cycles 3)))
                                 ;; If a multiple of three cycles have
                                 ;; elapsed, the fourth work period was
                                 ;; just completed, so take a longer break.
                                 "30 minutes"
                               "5 minutes"))
               :before (do (announce "Starting break time.")
                           (notify "Starting break time.")
                           (my-ticking-sound-cancel-timer))
               :advance (do (announce "Break time is over!")
                            (notify "Break time is over!"))))
    :after
    (do (my-ticking-sound-cancel-timer)))
```
 
  However, there I ran into issues that the timer keeps ticking if I exit the hammy without advancing, so I added the cancel-timer function as an after-advice to `hammy-stop` and to `hammy-mode-hook`:
  
  ``` emacs-lisp
  (add-hook 'hammy-mode-hook #'my-ticking-sound-cancel-timer)
  (define-advice hammy-stop (:after (&rest args))
    'my-ticking-sound-cancel-timer-advice
    (my-ticking-sound-cancel-timer))
```
(While writing this I looked at the source of `hammy-stop` again and realize that I might need to add the advice to `hammy-reset`, too.)

A separate feature-request from me might be to add a `hammy-stop-hook`, as a complement to `hammy-start-hook`, as all other hooks seem to be for when an interval/cycle/hammy is completed.

**Comment by alphapapa** [comment]
Hi Michael,

I think this feature would drive some users crazy, but I'm happy to support it for those who would like it.  :)  (Or, from another perspective, we just want to do something when the work period starts, and stop doing it when the work period ends--personally, I use a command that starts a music playlist of the length of the work interval, so when the music stops, I know it's break time.)

I've added the `hammy-stop-hook` as you requested.  Other than that, I'm not sure what we should do to support this in a built-in way, if anything.  We could add a `stop` slot to the `hammy` struct and call any functions in it when stopping the hammy.  That's probably a good idea, so I'll do that...

**Comment by alphapapa** [comment]
Ok, that's done, so now you can put your timer-canceling function in the hammy's `stopping` slot.  I still need to document all of the slots, but I'll probably do that after development has settled down a bit...

**Comment by meliache** [comment]
> think this feature would drive some users crazy
Yes, I can see that, it's not something that I would enable by default. Somehow I got used to it after a while and it's nice to hear that the interval is running even if emacs is in the background
>  I use a command that starts a music playlist of the length of the work interval, so when the music stops, I know it's break time.)
Oh, I never thought about that, it's a good idea, I used mpd so I could run `mpc-pause` and `mpc-play`. So for I often had both running, the pomodoro timer and the tick sound and the same time. which might drive people even more crazy.
> Ok, that's done, so now you can put your timer-canceling function in the hammy's stopping slot
Thanks, cool!
